packages/discord.js/src/structures/BaseChannel.js

```diff
@@ -25,6 +26,17 @@ class BaseChannel extends Base {
   }
 
   _patch(data) {
+    if ('flags' in data) {
+      /**
+       * The flags that are applied to the channel.
+       * <info>This is only `null` in a {@link PartialGroupDMChannel}. In all other cases, it is not `null`.</info>
+       * @type {?Readonly<ChannelFlagsBitField>}
+       */
+      this.flags = new ChannelFlagsBitField(data.flags).freeze();
+    } else {
+      this.flags = new ChannelFlagsBitField(this.flags).freeze();
```

Comment:
Shouldn't this be a default to `null`?
```suggestion
      this.flags ??= null;
```

Otherwise, if an instance is always desired, it should default initialize with `0`, as `this.flags` here is potentially `undefined`.

Reply:
Doesn't this have the same effect? Initialising with `undefined` returns a bit field of 0?

Reply:
Not necessarily, but I can see this giving us issues in some cases. For instance, `BitField`'s constructor defaults in the case of `undefined`, which may work in this case:

https://github.com/discordjs/discord.js/blob/669c3cd2566eac68ef38ab522dd6378ba761e8b3/packages/discord.js/src/util/BitField.js#L28-L34

However, this is a potentially nullable property, if we call `_patch` on `PartialGroupDMChannel`, it'll error. While this is not an immediate issue as we never patch them (given how those objects are retrieved), it may lead to issues in the future, be it from a new type of channel, us receiving GDMC update events, or even a bad translation/port in TSR if this is looked on for inspiration.

Nevertheless, all the 3 scenarios are incredibly unlikely. But we can do this instead to avoid cloning the bitfield instance (and also solves the aforementioned issue!):
```suggestion
      this.flags ??= new ChannelFlagsBitField().freeze();
```